libevm/LegacyVM.cpp

```diff
@@ -331,15 +331,19 @@ void LegacyVM::interpretCases()
             if (m_ext->staticCall)
                 throwDisallowedStateChange();
 
+            // Self-destructs only have gas cost starting with EIP 150
             m_runGas = toInt63(m_schedule->suicideGas);
-            Address dest = asAddress(m_SP[0]);
-
-            // After EIP158 zero-value suicides do not have to pay account creation gas.
-            if (m_ext->balance(m_ext->myAddress) > 0 || m_schedule->zeroValueTransferChargesNewAccountGas())
-                // After EIP150 hard fork charge additional cost of sending
-                // ethers to non-existing account.
-                if (m_schedule->suicideChargesNewAccountGas() && !m_ext->exists(dest))
-                    m_runGas += m_schedule->callNewAccountGas;
+
+            Address const dest = asAddress(m_SP[0]);
+            // Starting with EIP150, self-destructs need to pay both gas cost and account creation
+            // gas cost. Starting with EIP158, 0-value self-destructs don't need to pay this account
+            // creation cost.
+            if (m_schedule->zeroValueTransferChargesNewAccountGas() ||
```

Comment:
aleth-interpreter part is quite clear, but I'm not the fan of the change in LegacyVM.

I think we can assume that eip158 is always activated not earlier than eip150, and then simplify it to very similar to interpterter's version:
```
if (m_schedule->suicideChargesNewAccountGas())
{
  if (m_schedule->zeroValueTransferChargesNewAccountGas() || m_ext->balance(m_ext->myAddress) > 0)
  {
     if (!m_ext->exists(dest))
       m_runGas += m_schedule->callNewAccountGas;
  }
}
```

Reply:
Maybe add `assert(m_schedule->zeroValueTransferChargesNewAccountGas() || m_schedule->suicideChargesNewAccountGas())` - this was your first condition, and it should be always true when eip158 comes after eip150

Reply:
Agree that using eips would be simpler, but 
> This assumes that m_schedule->eip150Mode will not be true for the Spurious Dragon schedule.

this assumption is wrong, because `eip150Mode` is true for Tangerine Whistle and further, so including Spurious Dragon.

So I think it should be 
```
            if (m_schedule->eip150Mode && (!m_schedule->eip158Mode || m_ext->balance(m_ext->myAddress) > 0))
            {
                if (!m_ext->exists(dest))
                    m_runGas += m_schedule->callNewAccountGas;
            }
```

or leave it as 3 nested ifs as I suggested above similar to interpreter version.